databuilder/models/user.py

```diff
@@ -153,6 +164,31 @@ def get_user_node(self) -> GraphNode:
 
         return node
 
+    def get_user_record(self) -> RDSModel:
+        record_attr_map = {
+            RDSUser.email: self.email,
+            RDSUser.is_active: self.is_active,
+            RDSUser.first_name: self.first_name or '',
+            RDSUser.last_name: self.last_name or '',
+            RDSUser.full_name: self.full_name or '',
+            RDSUser.github_username: self.github_username or '',
+            RDSUser.team_name: self.team_name or '',
+            RDSUser.employee_type: self.employee_type or '',
+            RDSUser.slack_id: self.slack_id or '',
+            RDSUser.role_name: self.role_name or '',
+            RDSUser.updated_at: self.updated_at or 0
+        }
+
+        record = RDSUser(rk=User.get_user_model_key(email=self.email))
+        for attr, value in record_attr_map.items():
+            if value or not self.do_not_update_empty_attribute:
+                record.__setattr__(attr.key, value)
```

Comment:
could you add a comment on this part? Are you trying to set the user record for optional attribute?

Reply:
sure, I'll add a comment. Yes, it is going to set the empty attribute based on the flag